src/services/MessageAPI/index.js

```diff
@@ -0,0 +1,137 @@
+
+class MessageAPIService {
+    _sent_messages = new BehaviorSubject([]);
+    _received_messages = new BehaviorSubject([]);
+    _count = 0;
+
+    sent_messages$ = this._sent_messages.asObservable();
+    received_message$ = this._received_messages.asObservable();
+
+    sendMessage(message, fingerprint, signature, publicKey, sender_id, recipient_id, message_encryption_indicator_id) {
+        const query = `
+        mutation createMessageEncryptionIndicator($indicatorInput:CreateMessageEncryptionIndicatorInput!) {
+            createMessageEncryptionIndicator(input:$indicatorInput){
+                id
+                message
+                signature
+                fingerprint
+                publicKey
+                sender {
+                    id
+                }
+                recipient {
+                    id
+                }
+            }
+          }
+        `;
+        const variables = {
+            "messageInput": {
+                "message": message,
+                "clientMutationId": "1",
+                "fingerprint": fingerprint,
+                "signature": signature,
+                "publicKey": publicKey,
+                "senderId": sender_id,
+                "recipientId": recipient_id,
+                "messageEncryptionIndicatorId": message_encryption_indicator_id
+            }
+        };
+        response = this.__queryHandler(query, variables);
+        this.__addSentMessage(response);
+    }
+
+    checkMessages(recipientID) {
+        const query = `
+        query {
+            messages(recipient_id: $recipientID) {
+                id
+                message
+                signature
+                fingerprint
+                publicKey
+                sender {
+                    id
+                }
+                recipient {
+                    id
+                }
+            }
+        }
+        `;
+        const variables = {
+            "recipientID": recipientID,
+        };
+        response = this.__queryHandler(query, variables);
+        this.__populateReceivedMessages(response);
+    }
+
+    getSentMessages(senderID) {
+        const query = `
+        query {
+            messages(sender_id: $senderID) {
+                id
+                message
+                signature
+                fingerprint
+                publicKey
+                sender {
+                    id
+                }
+                recipient {
+                    id
+                }
+            }
+        }
+        `;
+        const variables = {
+            "senderID": senderID
+        };
+        response = this.__queryHandler(query, variables);
+        this.__populateSentMessages(response);
+    }
+
+    __populateReceivedMessages(data) {
+        data['data']['messages'].forEach(item => {
+            this._received_messages.next([
+                ...this._received_messages.value,
+                item
+            ]);
+        });
+    }
```

Comment:
```suggestion
    __populateReceivedMessages(data) {
          this._received_messages.next([
                ...this._received_messages.value,
                ...data['data']['messages']
            ]);
    }
```

Reply:
Ah you know what, that is an array. Why didn't I think of that.